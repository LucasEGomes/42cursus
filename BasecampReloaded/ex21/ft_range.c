#include <stdlib.h>

int	*ft_range(int min, int max)
{
	int				*result;
	unsigned int	size;
	unsigned int	index;

	if (min >= max)
		return (NULL);
	size = min + 1;
	size = max - size + 1;
	result = malloc(sizeof(*result) * ((size_t)size));
	if (result == NULL)
		return (NULL);
	index = 0;
	while (index < size)
	{
		result[index] = min + index;
		index++;
	}
	return (result);
}
