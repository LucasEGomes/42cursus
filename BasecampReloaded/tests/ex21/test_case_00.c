#include "ex21.h"

int	g_fake = 0;

int	main(void)
{
	int	expected_0[] = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9};
	int	expected_1[] = {4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16};
	
	run(0, 10, expected_0);
	run(4, 17, expected_1);
}
