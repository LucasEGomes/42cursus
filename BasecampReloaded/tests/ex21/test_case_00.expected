10|OK
13|OK
